tiledb/api/examples/using_tiledb_stats.rs

```diff
@@ -0,0 +1,117 @@
+extern crate tiledb;
+use tiledb::{Array, Result as TileDBResult};
+use tiledb::config::Config;
+use tiledb::vfs::VFS;
+use tiledb::Datatype;
+
+
+const ARRAY_NAME: &str = "stats_array";
+const ATTRIBUTE_NAME: &str = "a";
+
+pub fn create_array(row_tile_extent : u32, col_tile_extent : u32) -> TileDBResult<()> {
+    let tdb = tiledb::context::Context::new()?;
+    let config: Config = tiledb::config::Config::new()?;
+    let vfs: VFS = tiledb::vfs::VFS::new(&tdb, &config)?;
+
+    let is_cur_dir = vfs.is_dir(ARRAY_NAME)?;
+    if is_cur_dir {
+        vfs.remove_dir(ARRAY_NAME)?;
+    }
+
+    let domain = {
+        let rows: tiledb::array::Dimension =
+            tiledb::array::DimensionBuilder::new::<u32>(
+                &tdb,
+                "row",
+                Datatype::UInt32,
+                &[1, 12000],
+                &row_tile_extent,
+            )?
+            .build();
+        let cols: tiledb::array::Dimension =
+            tiledb::array::DimensionBuilder::new::<u32>(
+                &tdb,
+                "col",
+                Datatype::UInt32,
+                &[1, 12000],
+                &col_tile_extent,
+            )?
+            .build();
+
+        tiledb::array::DomainBuilder::new(&tdb)?
+            .add_dimension(rows)?
+            .add_dimension(cols)?
+            .build()
+    };
+
+    let attribute_a = tiledb::array::AttributeBuilder::new(
+        &tdb,
+        ATTRIBUTE_NAME,
+        tiledb::Datatype::Int32,
+    )?
+    .build();
+
+    let schema = tiledb::array::SchemaBuilder::new(
+        &tdb,
+        tiledb::array::ArrayType::Dense,
+        domain,
+    )?
+    .add_attribute(attribute_a)?
+    .build();
+
+    tiledb::Array::create(&tdb, ARRAY_NAME, schema)
+}
+
+pub fn write_array() -> TileDBResult<()> {
+    let tdb = tiledb::context::Context::new()?;
+    let array: Array = tiledb::Array::open(&tdb, ARRAY_NAME, tiledb::array::Mode::Write)?;
+    let mut data : Vec<i32> =  Vec::from_iter(0..12000 * 12000);
+
+    let query =
+        tiledb::QueryBuilder::new(&tdb, array, tiledb::QueryType::Write)?
+            .layout(tiledb::array::Layout::RowMajor)?
+            .dimension_buffer_typed(
+                ATTRIBUTE_NAME,
+                data.as_mut_slice(),
+            )?
+            .build();
+
+    query.submit().map(|_| ())
```

Comment:
Is this `.map(|_| ())` actually doing something important, or should it just be `query.submit()?` ?

Reply:
`query.submit()?;` might give an unused return value warning?

As I understood it there's only one query allowed to run on an open array at a time, so `query` consumes the array you pass it, and then my thinking was that `submit` should either return back `Self` if you can re-submit the query, or the array if not.  So in the current API at least there is a return value here.

Reply:
Hurms. I see the awkward there. Feel free to ignore this for now @abigalekim. We can clean it up when we figure out that query.submit API better since its not yet settled.

Reply:
Hm, as for now I just switched it to a question mark and added Ok(()) at the bottom of each function. I think this should suffice for now and makes more sense given the intention of the error messages.